libs/core/langchain_core/runnables/history.py

```diff
@@ -306,8 +306,23 @@ def get_session_history(
             history_chain = RunnablePassthrough.assign(
                 **{messages_key: history_chain}
             ).with_config(run_name="insert_history")
-        bound = (
+        bound_sync = (
```

Comment:
This patter is overly complex. 
It swaps between sync and async depending on whether the invoke was .invoke or .ainvoke

It may be easier to accept a parameter in the RunnableWithMessageHistory to specify whether we're using sync or async methods for the store